tests/test_network.py

```diff
@@ -36,19 +36,22 @@ def test_inference_run_with_restart():
     model = networks.Inference(
         model,
         center=center,
-        channels=channels,
         scale=scale,
         grid=schema.Grid.grid_720x1440,
-        channel_set=schema.ChannelSet.var34,
+        channel_names=["a", "b"],
     )
 
     step1 = []
-    for _, state, restart in model.run_steps_with_restart(x, 3):
+    time = datetime.datetime(2018, 1, 1)
```

Comment:
@yairchn @NickGeneva  This actually shows how the TimeLoop interface can be used if we modify the state in between. a bit messy, but we could clean it up. (#42)

Reply:
also #41

Reply:
Not sure I'm 100% following Noah. 

I do see you can add additional params into the inferencer call function.

The restart aspect is defined by the inferencer, which in many cases is model specific right. So you would need to define such capabilities on a model by model case unless you use the default TimeLoop.

You could modify these params to be some function or something. :thinking: